src/App.js

```diff
@@ -1,16 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const messageCopy = chatMessages.map((message) => {
+    return {
+      id: message.id,
+      sender: message.sender,
+      body: message.body,
+      timeStamp: message.timeStamp,
+      liked: message.liked,
+    };
+  });
+
+  let [messageData, setMessageData] = useState(messageCopy);
+
+  const toggleLike = (id) => {
+    let newMessageData = messageData.map((message) => {
+      if (message.id === id) {
+        message.liked = !message.liked;
+        return message;
+      } else {
+        return message;
+      }
+    });
+
+    setMessageData(newMessageData);
+  };
```

Comment:
Nice work! We can also refactor the conditional into a ternary as well:

```javascript
  const toggleLike = (id) => {
    let newMessageData = messageData.map((message) => 
       message.id === id? { ...message, liked: !message.liked } : message
    );

    setMessageData(newMessageData);
  };
```

Reply:
Nice work updating the state data by returning a brand new object with `map` and the spread operator!

Here's a refresher on updating state in React:

React components re-render whenever there is a change/update to state data. In this case, we are modifying a key-value pair within an object in a nested data structure.

Under the hood, React uses `Object.is()` to compare the previous state object with the one that's been provided via`setMessages(updatedMessages`). `Object.is()` checks if the object passed into the set state function has a different reference in memory. In Javascript, changing the properties and/or values in an object does NOT change the object reference, which is why we must create a new version of our state object (contains a copy of all the properties that weren't updated along with the properties/values that were).

Here is an article with more info: https://www.valentinog.com/blog/react-object-is/